### ISSUE TITLE
Deconstruct receiver ref escape is not considered

### ISSUE THREAD

**Issue by jjonescz** [dotnet/roslyn#75850]
**Version Used**: 4.13.0-2.24558.12 ([c0b2f36](https://github.com/dotnet/roslyn/commit/c0b2f366d0fb54cf49040a82f0d044bb959a9e73))

**Steps to Reproduce**:

```cs
using System.Diagnostics.CodeAnalysis;
class C
{
    R M(ref S s)
    {
        (var x, _) = s;
        return x; // error CS8352: Cannot use variable 'x' in this context because it may expose referenced variables outside of their declaration scope
    }
}
struct S
{
    [UnscopedRef] public void Deconstruct(out R x, out int y) => throw null;
}
ref struct R;
```

**Expected Behavior**: No error like in the equivalent code:

```cs
using System.Diagnostics.CodeAnalysis;
class C
{
    R M(ref S s)
    {
        s.Deconstruct(out var x, out _);
        return x; // ok
    }
}
struct S
{
    [UnscopedRef] public void Deconstruct(out R x, out int y) => throw null;
}
ref struct R;
```

**Actual Behavior**: An error is reported. We are analyzing the invocation whose receiver is a `BoundDeconstructValuePlaceholder` (those have always "current method" ref escape):

https://github.com/dotnet/roslyn/blob/543cb4568f28b0d2f2cfecdf2d56365b9252e848/src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs#L969

I think we could instead pass the original RHS of the deconstruction as the receiver for the analysis. (Assuming it's guaranteed the receiver will be passed as is and not copied while lowering/emitting.)

**Comment by jjonescz** [comment]
Turns out the lowered deconstruction copies the `s`, so the ref safety error is correct.
